server/src/main/java/org/opensearch/extensions/AddSettingsUpdateConsumerRequestHandler.java

```diff
@@ -68,25 +72,51 @@ TransportResponse handleAddSettingsUpdateConsumerRequest(AddSettingsUpdateConsum
 
                 // Extract setting and type from writeable setting
                 Setting<?> setting = extensionComponentSetting.getSetting();
+
+                // we need to get the actual setting from nodeSetting or indexsetting maps in SettingsModule
+                // use conditional based on setting properties
+                Setting<?> settingForUpdateConsumer = null;
+                if (setting.hasIndexScope()) {
+                    settingForUpdateConsumer = settingsModule.getIndexScopedSettings().get(setting.getKey());
+                } else if (setting.hasNodeScope()) {
+                    settingForUpdateConsumer = settingsModule.getClusterSettings().get(setting.getKey());
+                }
+                // do a null check and throw IllegalArgument exception here if neither index or node scope
```

Comment:
Stray comment needs to be removed but otherwise LGTM!

Reply:
Actually this comment indicates we need to do a null check and somehow set `status` to false to fix the failing tests.  Throwing an exception that gets caught below would work.